src/Core/Settings.h

```diff
@@ -772,6 +772,8 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Bool, output_format_json_array_of_rows, false, "Output a JSON array of all rows in JSONEachRow(Compact) format.", 0) \
     M(Bool, output_format_json_validate_utf8, false, "Validate UTF-8 sequences in JSON output formats, doesn't impact formats JSON/JSONCompact/JSONColumnsWithMetadata, they always validate utf8", 0) \
     \
+    M(String, format_json_object_each_row_column_for_object_name, "", "The name of column that will be used as object names in JSONObjectEachRow format. Column type should be String", 0) \
```

Comment:
```suggestion
    M(String, format_json_object_each_row_column_for_object_name, "", "The name of column that will be used as object keys in JSONObjectEachRow format. Column type should be String", 0) \
```

Reply:
I'm not sure, if this fix is entirely correct, because formally place where you insert value from a column is called "key" in JSON language, but feature is "object name", so feel free to discard it.